docs/data-guidelines/api.md

```diff
@@ -2,6 +2,49 @@
 
 This file contains guidelines that are specific to the web API features (`api/`).
 
+
+## Adding features and subfeatures
+
+A json file should be created for every _top level feature_ that has a released public implementation on one of the supported set of browsers (and not before).
+For Web APIs the top level features are individual Web API interfaces and the children of namespace objects like `Intl`.
+
+A possible structure for a Web API is shown below. 
+
+```
+api                           // category - always api for Web APIs
+  interface                   // top level feature
+    properties               // feature: required
+    methods                  // feature: required
+       parameters             //   subfeature: compatibility is different
+         parameter options    //   subfeature: if compatibility is different
+    support in workers     // subfeature: if compatibility is different (might also just be on specific methods)
+```
+
+The interface must include subnodes for _all_ its properties and methods (that have an implementation in at least one of the supported browsers), whether or not they were added at the same time as the parent interface, and hence share the same compatibility information.
```

Comment:
@foolip Part of the reason for this update was that the term subfeatures was undefined. I thought it just meant a sub node of some other node, but @queengooborg was using it to mean a property or a method below the top level. 

Sub-node on the other hand is unambiguous. I think we should use that term if we mean "create another level of nesting" unless we absolutely define subfeature to mean "nested definition at any level".

Reply:
@queengooborg can you clarify the definitions you'd like to use?

What we have in BCD is a tree, where some nodes in the tree have `__compat`. These nodes are called "features" in `traverse.js`, and I've often called them entries as well.

I think there are two potentially useful concepts:
- Any feature that sits below another one in the tree, like "api.AbortController.abort" (because "api.AbortController" is a feature)
- Things that are behaviors of another feature and only worth adding if support differs, like "api.fetch.blob_data_support".

I don't know what the arguments are, but my initial preference would be to just call everything features, and talk about "behavioral features" or "subfeatures" where needed to be precise.

Reply:
I'd like to define the terms as such:
- Feature: any object with a `__compat` statement (ex. api.Foo)
- Subfeature: any feature that is a level below another feature (ex. api.Foo.bar)
- Behavioral feature: any feature that describes a behavior, parameter, object, etc. (ex. api.Foo.worker_support, or api.Foo.bar.frozen_value)

Reply:
I think those definitions are good.